doc/adr/0016_widget_as_iFrame.md

```diff
@@ -0,0 +1,62 @@
+# 16. Widget as iFrame
+
+Date: 2023-01-10
+
+## Status
+
+Accepted
+
+## Context
+
+We want to provide a standardized entry point for the users in order for them to recognize
+the eID process across different eServices. We believe that a unified user-facing entry point, served as a widget,
+will benefit our customers. To achieve this, there are multiple options. We consider the following three as most feasible:
+
+1. provide framework-specific web components (e.g. react, vue) as a library
+
+   - benefits
+     - easy to import and easily reusable across eServices
+     - easy to ship updated components to the eServices
+     - highly reliable with expected behaviour within the framework
+   - costs
+     - difficult to integrate into legacy frontend stacks (likely to face in our governmental context)
+     - high maintenance / development effort from our side
+     - might over-complex a simple case
+
+2. bundle code into an iFrame and ship through a JavaScript <script> tag
+
+   - benefits
+     - low maintenance / development effort from our side, since one solution "fits all"
+     - easy to ship updated components to the eServices
+     - reliable provision, because JavaScript inside the iframe is running in the context of another page and main page rendering does not block the iFrame rendering
+     - no interference with main page’s CSS results in a standardized styling across eServices
+     - JavaScript's communication to main page is possible
+   - costs
+     - customers have to provide width and height values. Otherwise, the browser is rendering nothing
+     - clickjacking with iframes is possible, though can be reduced with X-FRAME-OPTIONS
```

Comment:
`Content-Security-Policy` (in this case `frame-ancestor`) instead of X-FRAME-OPTIONS